pkg/client/block/interface.go

```diff
@@ -0,0 +1,28 @@
+//go:generate mockgen -destination=../../../testutil/mockclient/block/block_client_mock.go -package=mockblockclient . Client
+package block
+
+import (
+	"context"
+
+	"github.com/pokt-network/poktroll/pkg/client"
+	"github.com/pokt-network/poktroll/pkg/observable"
+)
+
+type (
+	// Observable wraps the generic observable.ReplayObservable[Block] type
+	Observable observable.ReplayObservable[client.Block]
```

Comment:
I think this should be called `BlocksObservable` to improve clarity. I get that it's defined in the block package but when reading code that uses is in the block package, it can be a bit confusing.